observers/slack_resource.cpp

```diff
@@ -0,0 +1,82 @@
+#include <memory>
+#include <vector>
+
+#include "glog/logging.h"
+
+#include "mesos/mesos.hpp"
+
+#include "observers/slack_resource.hpp"
+
+#include "serenity/metrics_helpers.hpp"
+
+namespace mesos {
+namespace serenity {
+
+Try<Nothing> SlackResourceObserver::consume(const ResourceUsage& usage) {
+  std::unique_ptr<ExecutorSet> newSamples(new ExecutorSet());
```

Comment:
Could you elaborate? What kind of divergences do you see and how things could look different? I may have seen too little of Mesos code to feel it.